dask_match/expr.py

```diff
@@ -590,6 +630,16 @@ def _layer(self):
             (self._name, 0): (M.head, (self.frame._name, 0), self.n),
         }
 
+    def simplify(self):
+        if isinstance(self.frame, Elemwise):
+            operands = [
+                Head(op, self.n) if isinstance(op, Expr) else op
+                for op in self.frame.operands
+            ]
+            return type(self.frame)(*operands)
```

Comment:
@rjzamora I was thinking of your culling PR when writing this.  Maybe something like this would end up being easier?  I'm not sure.

Reply:
I think I agree that the `simplify` approach is more intuitive than a general replacement rule when you know that you will always want to push a particular expression down the tree as far as possible. I'll think a bit more about this, but I think I like the general idea.